src/pybind/mgr/dashboard/controllers/health.py

```diff
@@ -0,0 +1,190 @@
+            result['fs_map'] = self.fs_map()
+
+        if self._has_permissions(Permission.READ, Scope.OSD):
+            result['osd_map'] = self.osd_map()
+            result['scrub_status'] = self.scrub_status()
+            result['pg_info'] = self.pg_info()
+
+        if self._has_permissions(Permission.READ, Scope.MANAGER):
+            result['mgr_map'] = self.mgr_map()
+
+        if self._has_permissions(Permission.READ, Scope.POOL):
+            result['pools'] = self.pools()
+            result['df'] = self.df()
+            result['client_perf'] = self.client_perf()
+
+        if self._has_permissions(Permission.READ, Scope.HOSTS):
+            result['hosts'] = self.hosts()
+
+        if self._has_permissions(Permission.READ, Scope.RGW):
+            result['rgw'] = self.rgw()
+
+        if self._has_permissions(Permission.READ, Scope.ISCSI):
+            result['iscsi_daemons'] = self.iscsi_daemons()
+
+        return result
+
+    def basic_health(self):
+        health_data = mgr.get("health")
+        health = json.loads(health_data['json'])
+
+        # Transform the `checks` dict into a list for the convenience
+        # of rendering from javascript.
+        checks = []
+        for k, v in health['checks'].items():
+            v['type'] = k
+            checks.append(v)
+
+        checks = sorted(checks, key=lambda c: c['severity'])
+        health['checks'] = checks
+        return health
+
+    def client_perf(self):
+        result = CephService.get_client_perf()
+        if self._minimal:
+            result = self._partial_dict(
+                result,
+                ['read_bytes_sec', 'read_op_per_sec',
+                 'recovering_bytes_per_sec', 'write_bytes_sec',
+                 'write_op_per_sec']
+            )
+        return result
+
+    def df(self):
+        df = mgr.get("df")
+        df['stats']['total_objects'] = sum(
+            [p['stats']['objects'] for p in df['pools']])
+        if self._minimal:
+            df = dict(stats=self._partial_dict(
+                df['stats'],
+                ['total_avail_bytes', 'total_bytes', 'total_objects',
+                 'total_used_bytes']
+            ))
+        return df
+
+    def fs_map(self):
+        fs_map = mgr.get('fs_map')
+        if self._minimal:
+            fs_map = self._partial_dict(fs_map, ['filesystems', 'standbys'])
+            fs_map['standbys'] = [dict() for item in fs_map['standbys']]
+            fs_map['filesystems'] = [self._partial_dict(item, ['mdsmap']) for
+                                     item in fs_map['filesystems']]
+            for fs in fs_map['filesystems']:
+                mdsmap_info = fs['mdsmap']['info']
+                min_mdsmap_info = dict()
+                for k, v in mdsmap_info.items():
+                    min_mdsmap_info[k] = self._partial_dict(v, ['state'])
+                fs['mdsmap'] = dict(info=min_mdsmap_info)
+        return fs_map
+
+    def hosts(self):
```

Comment:
I thought it would make sense to name these methods after the keys that would be present in the return value of `all_health()`; is there a specific reason that you'd prefer this name change?

Reply:
Yes, because `hosts()` it doesn't return hosts at all, but just the number of hosts.